designs/sam-config.md

```diff
@@ -0,0 +1,282 @@
+SAM CLI App Level Config
+====================================
+
+
+What is the problem?
+--------------------
+
+Today users of SAM CLI need to invoke the CLI directly with all parameters supplied to its commands.
+
+for e.g: `sam build --use-container --debug`
+
+But often, during the lifecycle of building and deploying a serverless application. the same commands get run repeatedly to build, package and deploy, before solidifying into the final application. 
+
+These CLI commands are often long and have many changing parts.
+
+Have a look at the following series of workflows
+
+
+* `sam build --use-container --template ... --parameter-overrides=... --skip-pull-image --manifest ...`
+
+* `sam package --s3-bucket ... --template-file ... --output-template-file ... --s3-prefix ... --kms-key-id ...`
+
+* `sam deploy --template-file ... --stack-name ... --capabilities ... --tags ... --parameter-overrides ... --kms-key-id ...`
+
+If this could be condensed into a series of workflows that look like
+
+* `sam build`
+* `sam package`
+* `sam deploy`
+
+That would be a huge user experience win.
+
+What will be changed?
+---------------------
+
+The suite of commands supported by SAM CLI would be aided by looking for a configuration file thats locally located at the project root where template.yaml is located by default. 
+
+`.sam-app-config`
+
+This configuration would be used for specifiying the parameters that each of SAM CLI commands use and would be in TOML format.
+
+Running a SAM CLI command now automatically looks for `.sam-config` file and if its finds it goes ahead with parameter passthroughs to the CLI.
+
+```
+sam build
+Default Config file location: .sam-config
+..
+..
+..
+```
+
+
+The default location of a .sam-config can be replaced by overriding an environment variable called `SAM_CONFIG`
+
+`
+export SAM_CONFIG=~/Users/username/mysamconfig
+`
+
+Users can pass an identifier for the section that will be scanned within the configuration file to pass parameters through.
+
+Be default the `default` section of the configuration is chosen.
+
+```
+[default]
+
+[default.build]
+[default.build.paramaters]
+profile="srirammv"
+debug=true
+skip_pull_image=true
+use_container=true
+
+[default.package]
```

Comment:
Not overly familiar with TOML but is this required? What would we place in this section?